prov/gni/src/gnix_vc.c

```diff
@@ -1941,9 +1942,9 @@ static int __gnix_vc_push_work_reqs(struct gnix_vc *vc)
 			if ((ret != -FI_ENOSPC) &&
 			    (ret != -FI_EAGAIN)) {
 				/* TODO report error? */
-				GNIX_WARN(FI_LOG_EP_DATA,
-					  "Failed to push request %p: %s\n",
-					  req, fi_strerror(-ret));
+				GNIX_FATAL(FI_LOG_EP_DATA,
```

Comment:
This was a GNIX_WARN because it's errors outside of NOSPC and AGAIN are unexpected.  They shouldn't happen so it's all the same to me if we use GNIX_WARN of GNIX_FATAL.  To be fully compliant I think we would want to shut down this connection and send an EQ event here.